rust-wasm-id-allocator/distributed-id-allocator/tests/compressor_unit_tests.rs

```diff
@@ -393,15 +406,41 @@ fn test_cluster_expansion() {
     // Fill the initial cluster
     generate_n_ids(&mut compressor, 5);
     finalize_next_range(&mut compressor);
-    assert_eq!(compressor.get_telemetry_stats().expansion_count, 0);
-    assert_eq!(compressor.get_telemetry_stats().cluster_creation_count, 0);
+    let telemetry = compressor.get_telemetry_stats();
+    assert_eq!(telemetry.expansion_count, 0);
+    assert_eq!(telemetry.cluster_creation_count, 0);
 
     // Expand the initial cluster
     generate_n_ids(&mut compressor, 2);
     finalize_next_range(&mut compressor);
     assert_eq!(compressor.get_telemetry_stats().expansion_count, 1);
 }
 
+#[test]
+fn test_overflows_to_new_cluster() {
+    let mut compressor_a = IdCompressor::new();
+    let mut compressor_b = IdCompressor::new();
+    _ = compressor_a.set_cluster_capacity(3);
+
+    // Finalize initial cluster in compressor_a
+    generate_n_ids(&mut compressor_a, 2);
+    finalize_next_range(&mut compressor_a);
+    assert_eq!(compressor_a.get_telemetry_stats().cluster_creation_count, 1);
+
+    // Finalize new foreign cluster in compressor_a
+    generate_n_ids(&mut compressor_b, 2);
+    let range_b = compressor_b.take_next_range();
+    _ = compressor_a.finalize_range(&range_b);
+    assert_eq!(compressor_a.get_telemetry_stats().cluster_creation_count, 1);
+
+    // Overflow initial local cluster to new cluster
+    generate_n_ids(&mut compressor_a, 8);
+    finalize_next_range(&mut compressor_a);
+    let telemetry = compressor_a.get_telemetry_stats();
+    assert_eq!(telemetry.expansion_count, 0);
+    assert_eq!(telemetry.cluster_creation_count, 1);
```

Comment:
could assert about the eager final count on the stats object to be even more sure :)